plugins/test-management-resources/src/components/test-result/NextButton.svelte

```diff
@@ -0,0 +1,84 @@
+<!--
+// Copyright © 2024 Hardcore Engineering Inc.
+//
+// Licensed under the Eclipse Public License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License. You may
+// obtain a copy of the License at https://www.eclipse.org/legal/epl-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//
+// See the License for the specific language governing permissions and
+// limitations under the License.
+-->
+<script lang="ts">
+  import { onMount, onDestroy } from 'svelte'
+  import { Button, Loading, Location, navigate } from '@hcengineering/ui'
+  import { initializeIterator, testResultIteratorProvider, testIteratorStore } from './store/testIteratorStore'
+  import testManagement, { TestResult } from '@hcengineering/test-management'
+  import { Doc, type DocumentQuery, WithLookup } from '@hcengineering/core'
+  import { getClient } from '@hcengineering/presentation'
+  import { getObjectLinkFragment } from '@hcengineering/view-resources'
+  import view from '@hcengineering/view'
+
+  export let object: WithLookup<TestResult> | undefined
+  let isLoading = true
+  let hasNext = false
+  let navigateToNext = false
+
+  const client = getClient()
+  const hierarchy = client.getHierarchy()
+
+  testIteratorStore.subscribe(() => {
```

Comment:
I feel like this may produce leak of subscribers. The `subscribe` returns unsubscribe function that should be called to unsubscribe.

Reply:
Thank you, fixed